core/toolbox.js

```diff
@@ -296,10 +316,7 @@ Blockly.Toolbox.prototype.getClientRect = function() {
  * procedures.
  */
 Blockly.Toolbox.prototype.refreshSelection = function() {
-  var selectedItem = this.getSelectedItem();
-  if (selectedItem && selectedItem.getContents()) {
-    this.flyout_.show(selectedItem.getContents());
-  }
+  this.showAll();
```

Comment:
This means you'll scroll to the top of the toolbox every time you add a variable.  That seems wrong.  You still need to keep track of the selected category and scroll back to it, at a minimum.

Reply:
Weird- I'm not seeing this problem. ShowAll is not causing the scroll position to change.